syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -992,19 +992,57 @@ def torch_type(self):
         else:
             return self.child.torch_type()
 
-    def encrypt(self, public_key):
-        """This method will encrypt each value in the tensor using Paillier
-        homomorphic encryption.
+    def encrypt(self, encryption_method="mpc", **kwargs):
+        """
+        This method will encrypt each value in the tensor using Multi Party
+        Computation (default) or Paillier Homomorphic Encryption
 
         Args:
-            *public_key a public key created using
-                syft.frameworks.torch.he.paillier.keygen()
+            encryption_method (str): Currently supports 'mpc' for Multi Party
+                Computation and 'paillier' for Paillier Homomorphic Encryption
+            **kwargs:
+                With Respect to MPC accepts:
+                    workers (list): Parties involved in the sharing of the Tensor
+                    crypto_provider (syft.VirtualWorker): Worker responsible for the
+                        generation of the random numbers for encryption
+                    args_fix_prec (dict): To be parsed as a kwarg for the .fix_prec() method
+
+                With Respect to Paillier accepts:
+                    public_key (phe.paillier.PaillierPublicKey): Can be obtained using
+                        ```public_key, private_key = sy.frameworks.torch.he.paillier.keygen()```
+        Returns:
+            An encrypted version of the Tensor following the protocol specified
+
+        Raises:
+            NotImplementedError: If protocols other than the ones mentioned above are queried
+
         """
+        if encryption_method.lower() == "mpc":
+            args_fix_prec = kwargs.pop("args_fix_prec")
+            workers = kwargs.pop("workers")
+            crypto_provider = kwargs.pop("crypto_provider")
+
+            if args_fix_prec:
+                x_shared = self.fix_prec(**args_fix_prec).share(
+                    *workers, crypto_provider=crypto_provider, **kwargs
+                )
+            else:  # If no args for .fix_prec()
+                x_shared = self.fix_prec().share(
+                    *workers, crypto_provider=crypto_provider, **kwargs
+                )
+            return x_shared
+
+        elif encryption_method.lower() == "paillier":
+            x = self.copy()
+            x_encrypted = PaillierTensor().on(x)  # Instantiate the class
+            x_encrypted.child.encrypt_(kwargs["public_key"])  # Perform blacHomomorphic Encryption
```

Comment:
Extract public_key like you did for mpc (it's nicer :) ) 
and typo: blacHomomorphic